### ISSUE TITLE
Exception in 2019.3

### ISSUE THREAD

**Issue by nikitabobko** [JetBrains/ruby-type-inference#39]
```
java.lang.Throwable: org.jetbrains.plugins.ruby.settings.RubyTypeContractsSettings is registered as application service, but requested as project one
    at com.intellij.openapi.diagnostic.Logger.error(Logger.java:145)
    at com.intellij.serviceContainer.PlatformComponentManagerImpl.getService(PlatformComponentManagerImpl.kt:292)
    at com.intellij.openapi.components.ServiceManager.getService(ServiceManager.java:22)
    at org.jetbrains.plugins.ruby.ruby.codeInsight.stateTracker.RubyClassHierarchyWithCaching$Companion.getInstance(RubyClassHierarchyWithCaching.kt:121)
    at org.jetbrains.plugins.ruby.ruby.codeInsight.stateTracker.ClassHierarchySymbolProvider.processDynamicSymbols(ClassHierarchySymbolProvider.kt:19)
    at org.jetbrains.plugins.ruby.ruby.codeInsight.symbols.v2.ClassModuleSymbol.processChildren(ClassModuleSymbol.java:75)
    at org.jetbrains.plugins.ruby.ruby.codeInsight.symbols.structure.util.SymbolHierarchy.a(SymbolHierarchy.java:221)
    at org.jetbrains.plugins.ruby.ruby.codeInsight.symbols.structure.util.SymbolHierarchy.a(SymbolHierarchy.java:135)
    at org.jetbrains.plugins.ruby.ruby.codeInsight.symbols.structure.util.SymbolHierarchy.b(SymbolHierarchy.java:124)
    at com.intellij.openapi.util.RecursionManager$1.doPreventingRecursion(RecursionManager.java:113)
    at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:71)
    at org.jetbrains.plugins.ruby.ruby.codeInsight.symbols.structure.util.SymbolHierarchy.processSymbolChildren(SymbolHierarchy.java:123)
    at org.jetbrains.plugins.ruby.ruby.codeInsight.RubyOverrideImplementUtil.getSuperSymbolInScope(RubyOverrideImplementUtil.java:94)
    at org.jetbrains.plugins.ruby.ruby.codeInsight.RubyOverrideImplementUtil.getSuperMethods(RubyOverrideImplementUtil.java:74)
    at org.jetbrains.plugins.ruby.ruby.codeInsight.RubyOverrideImplementUtil.a(RubyOverrideImplementUtil.java:64)
    at org.jetbrains.plugins.ruby.ruby.codeInsight.RubyOverrideImplementUtil.getSuperMethods(RubyOverrideImplementUtil.java:50)
    at org.jetbrains.plugins.ruby.ruby.lang.search.overriden.RubySuperMethodsSearchExecutor.execute(RubySuperMethodsSearchExecutor.java:20)
    at org.jetbrains.plugins.ruby.ruby.lang.search.overriden.RubySuperMethodsSearchExecutor.execute(RubySuperMethodsSearchExecutor.java:15)
    at com.intellij.util.ExecutorsQuery.processResults(ExecutorsQuery.java:31)
    at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:91)
    at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:114)
    at com.intellij.util.UniqueResultsQuery.processResults(UniqueResultsQuery.java:38)
    at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:91)
    at com.intellij.util.AbstractQuery.forEach(AbstractQuery.java:83)
    at com.intellij.util.AbstractQuery.findAll(AbstractQuery.java:27)
    at org.jetbrains.plugins.ruby.ruby.codeInsight.lineMarkers.RubyLineMarkerProvider.a(RubyLineMarkerProvider.java:64)
    at org.jetbrains.plugins.ruby.ruby.codeInsight.lineMarkers.RubyLineMarkerProvider.collectSlowLineMarkers(RubyLineMarkerProvider.java:104)
    at com.intellij.codeInsight.daemon.impl.LineMarkersPass.queryProviders(LineMarkersPass.java:194)
    at com.intellij.codeInsight.daemon.impl.LineMarkersPass.lambda$doCollectInformation$3(LineMarkersPass.java:96)
    at com.intellij.codeInsight.daemon.impl.Divider.divideInsideAndOutsideInOneRoot(Divider.java:79)
    at com.intellij.codeInsight.daemon.impl.LineMarkersPass.doCollectInformation(LineMarkersPass.java:91)
    at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:55)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$null$1(PassExecutorService.java:429)
    at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:995)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:422)
    at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:592)
    at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:538)
    at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:421)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:397)
    at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:167)
    at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:162)
    at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:395)
    at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:161)
    at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
    at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
    at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
    at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
    at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

**Comment by nikitabobko** [comment]
https://youtrack.jetbrains.com/issue/RUBY-25022

**Comment by nikitabobko** [comment]
Fixed in v0.3.2
